src/Microsoft.AspNetCore.Routing/Matching/DefaultEndpointSelector.cs

```diff
@@ -73,8 +74,7 @@ public DefaultEndpointSelector(IEnumerable<MatcherPolicy> matcherPolicies)
             if (endpoint != null)
             {
                 feature.Endpoint = endpoint;
-                feature.Invoker = endpoint.Invoker;
-                feature.Values = values;
+                httpContext.Request.RouteValues = values;
```

Comment:
The passed in `IEndpointFeature` no longer has `RouteValues`.
What do you think of setting route values on the `HttpRequest` in `EndpointSelector`?